Source/DafnyCore/ProofDependencyWarnings.cs

```diff
@@ -32,7 +33,9 @@ public static void WarnAboutSuspiciousDependencies(DafnyOptions dafnyOptions, Er
       if (result.Outcome != VcOutcome.Correct) {
         continue;
       }
-      Warn(dafnyOptions, reporter, depManager, implementation.Name, result.VCResults.SelectMany(r => r.CoveredElements));
+      var assertCoverage = result.VCResults.Select(e => (e.CoveredElements, new HashSet<(Boogie.IToken Tok, string Description, string Id)>(e.Asserts))).ToList();
+      var unusedFunctions = UnusedFunctions(depManager, implementation.Name, result.VCResults.SelectMany(r => r.CoveredElements), result.VCResults.SelectMany(r => r.AvailableAxioms));
+      Warn(dafnyOptions, reporter, depManager, implementation.Name, assertCoverage, unusedFunctions);
```

Comment:
Can we give `Warn` a more specific name? Maybe rename it to `WarnAboutSuspiciousDependencies` and rename the original `WarnAboutSuspiciousDependencies` to `WarnAboutSuspiciousDependenciesUsingStoredPartialResults`